beacon-chain/db/filesystem/layout_flat.go

```diff
@@ -115,29 +114,26 @@ func (l *flatLayout) pruneBefore(before primitives.Epoch) (*pruneSummary, error)
 
 // Read slot from marshaled BlobSidecar data in the given file. See slotFromBlob for details.
 func slotFromFile(name string, fs afero.Fs) (primitives.Slot, error) {
```

Comment:
This is the main change, instead of reading from file offset with the assumption that it will be a blob file, we now change to unmarshaling the file, this is the only reliable way of determining which type it is (data column or blob).

There definitely will be a performance hit for this, however after chatting with Manu, we believe this is ok as 
1. we'll move to by-epoch soon which this won't be a problem (we'll get epoch by file path)
2. this only happens during startup or migration, which shouldn't impact syncing perf